numpy/add_newdocs.py

```diff
@@ -3060,8 +3060,10 @@ def luf(lamdaexpr, *args, **kwargs):
     """
     Number of elements in the array.
 
-    Equivalent to ``np.prod(a.shape)``, i.e., the product of the array's
-    dimensions.
+    Equal to ``np.prod(a.shape)``, i.e., the product of the array's
+    dimensions, except that ``size`` returns an instance of ``int``
+    whereas ``np.prod(a.shape)`` returns an instance of ``np.int_``
+    (e.g. ``np.int64``).
```

Comment:
I think this obscures the information being provided a little. I'd suggest keeping the original sentence unchanged, perhaps with `s/Equivalent/Equal/`, and then add the type observation as a separate paragraph, or even in a Notes section. For most users the distinction is not really relevant, so I don't think it deserves equal footing with the much more useful "here's how this value is calculated."

I would prefer something along the lines of this:

```
Notes
-----
`a.size` returns a standard arbitrary precision Python integer. This 
may not be the case with other methods of obtaining the same value
(like the suggested ``np.prod(a.shape)``, which returns an instance
of ``np.int_``), and may be relevant if the value is used further in
calculations that may overflow a fixed size integer type.
```